### DISCUSSION TITLE
[Other] Are versions of RabbitMQ prior to 3.9.17 compatible with RHEL 9?

### DISCUSSION THREAD

**Discussion by victorkeanework** [rabbitmq/rabbitmq-server#12836] [Other]
### Community Support Policy

- [X] I have read [RabbitMQ's Community Support Policy](https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md)

### RabbitMQ version used

3.13.7 or older

### How is RabbitMQ deployed?

RPM package

### Steps to reproduce the behavior in question

Versions of RabbitMQ from version 3.9.17 onwards stopped supporting CentOS 7 and RHEL 7 - https://github.com/rabbitmq/rabbitmq-server/discussions/4637

It was stated that Erlang RPM packages of Erlang 24.3 produced by our team
now provide packages compatible with Fedora, Rocky Linux, CentOS Stream 8,
 and Centos Stream 9." which seems to imply that they weren't supported before due to the OpenSSL version.

However there were version of Erlang 24.3 that did support CentOS 7 - "Regular CentOS 7 and Amazon Linux 2 builds were produced up to Erlang 23.3.4.18" - https://github.com/rabbitmq/erlang-rpm?tab=readme-ov-file

For example RabbitMQ 3.9.16 used Erlang 23.3. Was this supported on both RHEL 7 and RHEL 9?

The documentation for Erlang 22.3.x states that it suports CentOS 8 - https://github.com/rabbitmq/erlang-rpm/tree/erlang-22.3.x

I've read a different question which states "The CentOS Stream 8/RHEL 8 package works perfectly fine on CentOS Stream 9 and RHEL 9". Does that mean that any version using Erlang 22.3.x, is supported by both RHEL 7 and RHEL 9 due to its support by CentOS 8? - https://github.com/rabbitmq/rabbitmq-server/discussions/8554

**Comment by michaelklishin** [comment]
RabbitMQ 3.9 has been [out of support](https://www.rabbitmq.com/release-information) for a few years now. You will not receive any more responses to your questions until you upgrade to 4.0.x

RabbitMQ is architecture and OS-neutral. As long as you can run a [compatible Erlang version](https://www.rabbitmq.com/docs/which-erlang#old-timers), you can run RabbitMQ.

RabbitMQ 3.9.29 supports Erlang versions between `24.3.4.2` and `25.2.x`. Have a look at this [final `25.2.x` Erlang release](https://github.com/rabbitmq/erlang-rpm/releases/tag/v25.2.3) and you will find RPMs built on and for CentOS Stream 9.

Even CentOS Stream 8 RPMs would likely run on Stream 9.

**Comment by michaelklishin** [comment]
"CentOS 7 is not supported" means that our team will not investigate any issues specific to CentOS 7. It is a dead distribution whose repositories were taken offline. I mean, that sort of sends a very strong message that it really is done, entirely unsupported, and no one should target it.

It also means that should our current way of producing CentOS 7 RPMs of cutting edge Erlang with a statically linked OpenSSL 1.1.x build break, we will immediately stop building those packages. Consider anything CentOS 7-related that you can get from our team an act of good will. I mean, you can run RabbitMQ `4.0.4` on Erlang 26.x (maybe even 27.x, I don't recall if we have built the CentOS 7 variant) on a distribution that Red Hat have completely abandoned over a year ago.
